filter_plugins/openshift_version.py

```diff
@@ -46,50 +46,82 @@ def _gte_function(version, deployment_type):
+
+
 # pylint: disable=too-few-public-methods
 class FilterModule(object):
     """
     Filters for version checking.
     """
-    #: The major versions to start incrementing. (enterprise, origin)
-    majors = [(3, 1)]
-
-    #: The minor version to start incrementing
-    minor = 3
-    #: The number of iterations to increment
-    minor_iterations = 10
+    # The major versions to start incrementing.
+    # Each element is composed of (major, minor_start, minor_end)
+    # Origin began versioning 3.x with 3.6, so begin 3.x with 3.6.
+    versions = [(3, 6, 10)]
 
     def __init__(self):
         """
         Creates a new FilterModule for ose version checking.
         """
         self._filters = {}
-        # For each major version
-        for enterprise, origin in self.majors:
+
+        # For each set of (major, minor, minor_iterations)
+        for major, minor_start, minor_end in self.versions:
             # For each minor version in the range
-            for minor in range(self.minor, self.minor_iterations):
+            for minor in range(minor_start, minor_end):
                 # Create the function name
-                func_name = 'oo_version_gte_{}_{}_or_{}_{}'.format(
-                    enterprise, minor, origin, minor)
+                func_name = 'oo_version_gte_{}_{}'.format(major, minor)
                 # Create the function with the builder
-                func = gte_function_builder(
-                    func_name, {
-                        'enterprise': '{}.{}.0'.format(enterprise, minor),
-                        'origin': '{}.{}.0'.format(origin, minor)
-                    })
+                func = gte_function_builder(func_name, "{}.{}.0".format(major, minor))
                 # Add the function to the mapping
                 self._filters[func_name] = func
 
-        # Create filters with special versioning requirements
-        self._filters['oo_version_gte_3_1_or_1_1'] = gte_function_builder('oo_version_gte_3_1_or_1_1',
-                                                                          {'enterprise': '3.0.2.905',
-                                                                           'origin': '1.1.0'})
-        self._filters['oo_version_gte_3_1_1_or_1_1_1'] = gte_function_builder('oo_version_gte_3_1_or_1_1',
-                                                                              {'enterprise': '3.1.1',
-                                                                               'origin': '1.1.1'})
-        self._filters['oo_version_gte_3_2_or_1_2'] = gte_function_builder('oo_version_gte_3_2_or_1_2',
-                                                                          {'enterprise': '3.1.1.901',
-                                                                           'origin': '1.2.0'})
+        # Create filters with special versioning requirements.
+        # Treat all Origin 1.x as special case.
+        legacy_filters = [{'name': 'oo_version_gte_3_1_or_1_1',
```

Comment:
I am unsure about something: is this **data** we could extract from the **code**? Is it possible that we could update an external data source (some kind of serialized file, json, yaml, w/e) and read from that instead? Would that be too difficult/out of scope of this PR?

I bring this up because I'm a fan of separating data from code. You really shouldn't have to update code files to update data in an *ideal* situation. I understand if that's not something we can do right now though. Ansible is its own beast.

Same comment applies to `test/openshift_version_tests.py:OpenShiftVersionTests.legacy_gte_tests`.